spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/repository/query/SpannerStatementQueryExecutor.java

```diff
@@ -188,7 +188,18 @@ else if (part.shouldIgnoreCase() != IgnoreCaseType.NEVER) {
 
 					switch (part.getType()) {
 					case LIKE:
-						andString += " LIKE %" + insertedTag;
+						andString += " LIKE " + insertedTag;
+						break;
+					case NOT_LIKE:
+						andString += " NOT LIKE " + insertedTag;
+						break;
+					case CONTAINING:
+						andString = " REGEXP_CONTAINS(" + andString + "," + insertedTag
```

Comment:
I'm not sure what `CONTAINING` should mean, but Ray said `LIKE %search%`, not REGEX.

Reply:
Turns out we cannot just append and prepend the % symbols. Those symbols must be part of the actual param. We have not laid out this code to allow actual changing of the params, which Spring Data provides to us as  Objects. Therefore, unless we want to make things hacky and ugly, and unless there is a straight up CONTAINS(string) function from Cloud Spanner, this REGEXP_CONTAINS is the best choice.